OpenOversight/app/main/views.py

```diff
@@ -74,26 +76,37 @@ def submit_complaint():
                            officer_image=request.args.get('officer_image'))
 
 
-@main.route('/submit')
+@main.route('/submit', methods=['GET', 'POST'])
+@login_required
 def submit_data():
-    return render_template('submit.html')
-
-
-@main.route('/upload', methods=['POST'])
-def upload_file():
-    file = request.files['file']
-    if file and allowed_file(file.filename):
-        filename = secure_filename(file.filename)
-        file.save(os.path.join(current_app.config['UNLABELLED_UPLOADS'], filename))
-        return redirect(url_for('main.show_upload',
-                                filename=filename))
-
-
-@main.route('/show_upload/<filename>')
-def show_upload(filename):
-    # return send_from_directory('../'+config.UNLABELLED_UPLOADS,
-    #                           filename)
-    return 'Successfully uploaded: {}'.format(filename)
+    form = HumintContribution()
+    if request.method == 'POST' and form.validate_on_submit():
+        filename = secure_filename(request.files[form.photo.name].filename)
+        image_data = request.files[form.photo.name].read()
+
+        # See if there is a matching photo already in the db
+        hash_img = hash_file(image_data)
+        hash_found = Image.query.filter_by(hash_img=hash_img).first()
+
+        if not hash_found:
```

Comment:
Addressed the first comment in commit f7bd0f7 by never using the original filename and always using the hash of the file to store both on the webserver filesystem and on S3.

Addressed this in commit ab636de to use `tempfile` to create a temporary directory and then store the file in there, again never using the original filename from the user (I followed [this guide](https://security.openstack.org/guidelines/dg_using-temporary-files-securely.html) to do so).